tests/unit/s2n_build_test.c

```diff
@@ -65,6 +65,10 @@ int main()
         return 0;
```

Comment:
Rather than adding another toggle/conditional check to things, could we create a symlink to the default libcrypto that plays nicely with the current `/usr/local/$LIBCRYPTO` structure? Even if libcrypto in this case is simply called "default"?

It's just that as soon as we start throwing different path structure in then we lose the ability to write simpler buildspecs like this one: https://github.com/jmayclin/s2n-tls/blob/cmake-asan-build/codebuild/spec/buildspec_asan.yml#L50-L63

Reply:
I'll also be honest and say that I have a lot of trouble following our setup_env and install_deps scripts, which is why I have generally gone for just writing the super boring buildspecs with no dependencies other than the docker image.

Reply:
This test uses the S2N_LIBCRYPTO variable set by the CI, so it's not really directly related to symlinks. The test wants to make sure we actually used the libcrypto the CI asked for, but if the CI didn't ask for a libcrypto, there's nothing to check.

For path structure: I could look into it, but that would require knowing where the default libcrypto actually is so that we can link to it. At the moment, that's handled by find_package in our cmake. I'd hate to have to do it manually, considering the fun variability between environments.

Reply:
Maybe ldconfig is portable enough?

Reply:
> I'll also be honest and say that I have a lot of trouble following our setup_env and install_deps scripts, which is why I have generally gone for just writing the super boring buildspecs with no dependencies other than the docker image.

I guess good news then, because "default" just bypasses setup_env and is only required in any particular install_deps to skip whatever strange install you planned to do. We just leave libcrypto to cmake.

Reply:
Nice, cmake is smart enough that it works just fine with that buildspec: [link](https://us-west-2.console.aws.amazon.com/codesuite/codebuild/024603541914/projects/AddressSanitizer/batch/AddressSanitizer%3A134ca6d0-0368-4112-b7f7-2ea70e04d4be/builds?region=us-west-2)
```
-- Found crypto: /usr/lib/x86_64-linux-gnu/libcrypto.a  
-- LibCrypto Include Dir: /usr/include
-- LibCrypto Shared Lib:  /usr/lib/x86_64-linux-gnu/libcrypto.so
-- LibCrypto Static Lib:  /usr/lib/x86_64-linux-gnu/libcrypto.a
```
Apparently cmake doesn't care if you give it a CMAKE_PREFIX_PATH that doesn't exist.